src/integrationtest/java/com/ericsson/gerrit/plugins/eiffel/integrationtest/ServiceIntegrationSteps.java

```diff
@@ -0,0 +1,282 @@
+            throws URISyntaxException, ClientProtocolException, IOException {
+        String jsonBodyString = "{\"project\":\"" + projectName + "\" ,"
+                + "\"subject\" : \"Let's support 100% Gerrit workflow direct in browser\", "
+                + "\"topic\" : \"create-change-in-browser\", " + "\"status\" : \"DRAFT\", "
+                + "\"branch\":\"master\"}";
+
+        HttpRequest request = new HttpRequest(HttpMethod.POST);
+        ResponseEntity response;
+        response = request.setHeader("Cookie", gerritAccountCookie)
+                .setHeader("Content-Type", "application/json; charset=UTF-8")
+                .setHeader("Accept", "application/json")
+                .setHeader("X-Gerrit-Auth", gerritXAuthToken).setBaseUrl(GERRIT_BASE_URL)
+                .setEndpoint("/changes/").setBody(jsonBodyString).performRequest();
+
+        return response;
+    }
+
+    protected List<String> consumeMessages(int messageCount, long timeout)
+            throws IOException, TimeoutException {
+        List<String> messages = new ArrayList<String>();
+        ConnectionFactory factory = new ConnectionFactory();
+        factory.setHost(RABBITMQ_HOSTNAME);
+        factory.setPort(RABBITMQ_PORT);
+        factory.setUsername(RABBITMQ_USERNAME);
+        factory.setPassword(RABBITMQ_PASSWORD);
+        Connection connection = factory.newConnection();
+        Channel channel = connection.createChannel();
+
+        long stopTime = System.currentTimeMillis() + timeout;
+
+        while (System.currentTimeMillis() < stopTime) {
+            try {
+                GetResponse response = channel.basicGet(RABBITMQ_QUEUENAME, true);
+                if (response != null) {
+                    messages.add(new String(response.getBody(), "UTF-8"));
+                }
+
+                if (messages.size() == messageCount) {
+                    return messages;
+                }
+            } catch (Exception e) {
+                LOGGER.error("RabbitMQ failed to get from queue", e);
+            }
+        }
+
+        return messages;
+    }
+
+    protected void declareQueues() throws IOException, TimeoutException {
+        ConnectionFactory factory = new ConnectionFactory();
+        factory.setHost(RABBITMQ_HOSTNAME);
+        factory.setPort(RABBITMQ_PORT);
+        factory.setUsername(RABBITMQ_USERNAME);
+        factory.setPassword(RABBITMQ_PASSWORD);
+        Connection connection = factory.newConnection();
+        Channel channel = connection.createChannel();
+
+        final AMQP.Exchange.DeclareOk exchangeOK = channel.exchangeDeclare(RABBITMQ_EXCHANGE_NAME,
+                "topic", true);
+        final AMQP.Queue.DeclareOk queueOK = channel.queueDeclare(RABBITMQ_QUEUENAME, true, false,
+                false, null);
+        final AMQP.Queue.BindOk bindOK = channel.queueBind(RABBITMQ_QUEUENAME,
+                RABBITMQ_EXCHANGE_NAME, "#");
+        assertEquals(true, exchangeOK != null);
```

Comment:
Please add messages to the assets so if it fails you dont just get expected true but was false.

Reply:
done